platform/api.htmlui/src/org/netbeans/spi/htmlui/HtmlViewer.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.spi.htmlui;
+
+import java.net.URL;
+import java.util.concurrent.Callable;
+import java.util.function.Consumer;
+
+/**
+ *
+ * @param <HtmlView>
+ * @since 1.21
+ */
+public interface HtmlViewer<HtmlView> {
```

Comment:
The new SPI between HTML UI API and the VSCode module to do the "viewing" differently.

Reply:
Integrated by  #3357